docassemble/ALWeaver/interview_generator.py

```diff
@@ -917,6 +917,9 @@ def get_docx_variables( text:str )->set:
   # Variables in very simple `if` statements (allow paragraph and whitespace flags)
   for possible_variable in re.findall(r'{%[^ \t]* +if ([^\} ]+) +[^ \t]*%}', text):
     minimally_filtered.add( possible_variable )
+  # Capture variables in `if` statements that contain a comparison
+  for possible_variable in re.findall(r'{%[^ \t]* +if ([^\} ]+) ==|is|>|<|!=|<=|>= .* +[^ \t]*%}', text):  
```

Comment:
We're starting to get into esoteric territory here, but there's some improvements here that we can make to accept additional valid jinja templates (https://jinja.palletsprojects.com/en/3.0.x/templates/):
* allow `+` and `-` directly before a `%` at the end or after a `%` at the beginning, no whitespace (disables / enables trim new lines)
*  should check for `elseif` statements too, more variables can be in there
* `in` is a valid operator too
* this does assume that the variable name is always the first item in the expression, when it could be the second. We'd have to start parsing and recognizing literals though, and that might be a bit much. We're starting down a "trying to use regex to parse unrestricted HTML" path

Reply:
Some of these should be easy changes, good thoughts